File: Selenium/Activities/Activity17.java
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity17 {
	public static void main(String args[]) {
		WebDriver driver;
		driver = new FirefoxDriver();
		driver.get("https://training-support.net/webelements/selects");
		System.out.println("The title of the page is: "+driver.getTitle());
		
		WebElement element = driver.findElement(By.cssSelector(".h-80"));
		Select multiSelect = new Select(element);
		
		multiSelect.selectByVisibleText("HTML");
		
		for (int i = 3; i <= 5; i++) {
            multiSelect.selectByIndex(i);
        }
		
		multiSelect.selectByValue("nodejs");
		
		List<WebElement> selectedoptions = multiSelect.getAllSelectedOptions();
		System.out.println("selected options are: ");
		for(WebElement options:selectedoptions) {
			System.out.println(options.getText());
		}
		
		
		multiSelect.deselectByIndex(4);
	
		selectedoptions = multiSelect.getAllSelectedOptions();
				System.out.println("selected options are: ");
				for(WebElement options:selectedoptions) {
					System.out.println(options.getText());
				}
				
		driver.quit();
		
	}

}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.select import Select
with webdriver.Firefox() as driver:
    wait=WebDriverWait(driver,timeout=10)
    driver.get("https://training-support.net/webelements/selects")
    print("the title of the page is: " ,driver.title)

    element = driver.find_element(By.CSS_SELECTOR,".h-80")
    multiSelect = Select(element)

    multiSelect.select_by_visible_text("HTML")
    for i in range(3,6):
       multiSelect.select_by_index(i)

    multiSelect.select_by_value("nodejs")

    selectedOptions = multiSelect.all_selected_options
    print("selected options are: ")
    for option in selectedOptions:
        print(option.text)

    multiSelect.deselect_by_index(4)

    
    selectedOptions = multiSelect.all_selected_options
    print("selected options are: ")
    for option in selectedOptions:
        print(option.text)

    driver.quit()